apple/internal/partials/debug_symbols.bzl

```diff
@@ -310,6 +310,13 @@ def _bundle_dsym_files(
             found_binaries_by_arch = found_binaries_by_arch,
             platform_prerequisites = platform_prerequisites,
         )
+        dsyms_command = (" && ".join([
+            "cp \"{dsym_path}\" \"${{OUTPUT_DIR}}/Contents/Resources/DWARF/{dsym_bundle_name}\"".format(
+                dsym_path = dsym.path,
+                dsym_bundle_name = dsym_output_filename,
+            )
+            for dsym in output_files
+        ]))
 
```

Comment:
This seems to break other tests/builds but it got me the DWARF binaries from `AppleDsymBundleInfo` copied over correctly when using the `xcarchive` rule.

Before:

```
bazel-bin/Code/Apps/CashApp/Cash.xcarchive/dSYMS
├── Cash.app.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
├── Siri UI.appex.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
├── Siri.appex.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
└── Widget Extension.appex.dSYM
    └── Contents
        ├── Info.plist
        └── Resources
            └── DWARF
```

After:

```
bazel-bin/Code/Apps/CashApp/Cash.xcarchive/dSYMS
├── Cash.app.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
│               └── Cash
├── Siri UI.appex.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
│               └── Siri UI
├── Siri.appex.dSYM
│   └── Contents
│       ├── Info.plist
│       └── Resources
│           └── DWARF
│               └── Siri
└── Widget Extension.appex.dSYM
    └── Contents
        ├── Info.plist
        └── Resources
            └── DWARF
                └── Widget Extension
```

@keith Would you happen to know how the `AppleDsymBundleInfo` provider is intended to be used? We use it in [xcarchive rule](https://github.com/bazelbuild/rules_apple/blob/afd97b7f9e0530b6848ed45844febe8de21bdcb7/apple/internal/xcarchive.bzl#L36-L39) and from what I can tell, before this change, the paths provided in that provider would only contain the `Info.plist` at the time of execution.

Reply:
From what I can tell the `AppleDebugInfo` provider correctly gives us the binary paths, the only difference is that `AppleDsymBundleInfo` seems to add a `/dSYMS/` directory path:

```
AppleDsymBundleInfo: bazel-out/<sha>/bin/Code/Apps/CashAppWidgets/WidgetExtension/dSYMs/WidgetExtension_dsyms/Widget Extension.appex.dSYM
AppleDebugInfo: bazel-out/<sha>/bin/Code/Apps/CashAppWidgets/WidgetExtension/WidgetExtension_dsyms/Widget Extension.appex.dSYM
```

It looks like thats intentional given: https://github.com/bazelbuild/rules_apple/blob/afd97b7f9e0530b6848ed45844febe8de21bdcb7/apple/internal/partials/debug_symbols.bzl#L330-L332

But then the rule copies it without the `/dSYMS`: https://github.com/bazelbuild/rules_apple/blob/afd97b7f9e0530b6848ed45844febe8de21bdcb7/apple/internal/partials/debug_symbols.bzl#L132-L137